src/main/java/firebase/sdk/dashboard/dao/UserDaoDatastore.java

```diff
@@ -0,0 +1,73 @@
+package firebase.sdk.dashboard.dao;
+
+import firebase.sdk.dashboard.data.SDK;
+import firebase.sdk.dashboard.data.SDKReleaseMetadata;
+import firebase.sdk.dashboard.data.VersionMetadata;
+import firebase.sdk.dashboard.data.Release;
+import firebase.sdk.dashboard.data.Platform;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.time.Instant;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultIterable;
+
+/* Class to implement UserDao */
+
+public class UserDaoDatastore implements UserDao {
+
+    public static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
+
+    // Get a user from the datastore
+    public User getUser(String id){
+        FilterPredicate userPropertyFilter = makePropertyFilter("id", id);
+        Query query = new Query("User").setFilter(userPropertyFilter);
+
+        PreparedQuery preparedQuery = DATASTORE.prepare(query);
+        FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
+        Entity user = preparedQuery.asSingleEntity();
+        if (user == null) {
+            return null;
+        }
+        return user;
+    }
+
+    // Add a user to the datastore
+    public void addUser(User user){
+       Key userKey = KeyFactory.createKey("User",user.uid());
+       
+        // Create a user entity
+        Entity userEntity = new Entity(userKey);
+        userEntity.setProperty("id",user.uid());
+        userEntity.setProperty("email",user.email());
+        userEntity.setProptery("favoriteSDKs",user.favoriteSDKs);
+
+        DATASTORE.put(userEntity);
+    }
+ 
+    // Updates user favorite SDKs
+    public void updateUser(User user){
+         User outdated = getUser(user.uid());
+         outdated.favoriteSDKs() = user.favoriteSDKs();
+         addUser(User outdated);
```

Comment:
This will not compile. Remove User and also try compiling the code to catch more such issues.